guvnor-structure/guvnor-structure-backend/src/main/java/org/guvnor/structure/backend/repositories/PullRequestServiceImpl.java

```diff
@@ -0,0 +1,265 @@
+                                   final long id ) {
+
+        checkNotEmpty( "repository", repository );
+
+        GitMetadata metadata = getRepositoryMetadata( repository );
+        PullRequest pullRequest = metadata.getPullRequest( id );
+
+        List<PullRequest> finalPullRequests = metadata.getPullRequests();
+        finalPullRequests.remove( pullRequest );
+
+        GitMetadata storableMetadata = new GitMetadataImpl( metadata.getName(), metadata.getOrigin(), metadata.getForks(), finalPullRequests );
+        this.metadataStore.write( storableMetadata.getName(), storableMetadata );
+
+        this.deleteHiddenBranch( pullRequest );
+
+    }
+
+    @Override
+    public List<FileDiff> diff( final PullRequest pullRequest ) {
+
+        final Repository repository = configuredRepositories.getRepositoryByRepositoryAlias( pullRequest.getTargetRepository() );
+        String diff = String.format( "diff:%s,%s", pullRequest.getTargetBranch(), pullRequest.getSourceBranch() );
+        return (List<FileDiff>) this.ioService.readAttributes( convert( repository.getRoot() ), diff );
+    }
+
+    public void changeStatusPullRequest( final String repository,
+                                         final long id,
+                                         final PullRequestStatus status ) {
+
+        checkNotEmpty( "repository", repository );
+        checkNotNull( "status", status );
+
+        GitMetadata metadata = getRepositoryMetadata( repository );
+        PullRequest pullRequest = metadata.getPullRequest( id );
+        PullRequestImpl finalPullRequest = new PullRequestImpl( pullRequest.getId(),
+                                                                pullRequest.getSourceRepository(),
+                                                                pullRequest.getSourceBranch(),
+                                                                pullRequest.getTargetRepository(),
+                                                                pullRequest.getTargetBranch(),
+                                                                status );
+
+        List<PullRequest> finalPullRequests = metadata.getPullRequests( elem -> elem.getId() != id );
+        finalPullRequests.add( finalPullRequest );
+
+        GitMetadata storableMetadata = new GitMetadataImpl( metadata.getName(), metadata.getOrigin(), metadata.getForks(), finalPullRequests );
+        this.metadataStore.write( storableMetadata.getName(), storableMetadata );
+
+    }
+
+    private GitMetadata getRepositoryMetadata( final String repository
+    ) {
+        final Optional<GitMetadata> optional = this.metadataStore.read( repository );
+
+        if ( !optional.isPresent() ) {
+            throw new RepositoryNotFoundException(
+                    String.format( "The repository does not exists <<%s>>", repository ) );
+        }
+
+        return optional.get();
+    }
+
+    private long generatePullRequestId( final GitMetadata metadata ) {
```

Comment:
This is interesting, since ```PullRequests``` are held at the ```Repository``` level and multiple Users can be working on the same ```Repository``` should this method not be synchronised?